typescript/packages/plugins/ironclad/src/ironclad.service.ts

```diff
@@ -0,0 +1,796 @@
+import { Tool } from "@goat-sdk/core";
+import { EVMWalletClient } from "@goat-sdk/wallet-evm";
+import { formatUnits, parseUnits } from "viem";
+import type { Address } from "viem";
+import { z } from "zod";
+import { ERC20_ABI } from "./abi/erc20";
+import { LENDING_POOL_ABI } from "./abi/lendingPool";
+import { PROTOCOL_DATA_PROVIDER_ABI } from "./abi/protocolDataProvider";
+import { LoopDepositParameters, LoopWithdrawParameters } from "./parameters";
+import {
+    BorrowIUSDParameters,
+    RepayIUSDParameters,
+    MonitorPositionParameters,
+} from "./parameters";
+import { CalculateMaxWithdrawableParameters } from "./parameters";
+
+interface LoopPosition {
+    borrowedAmounts: string[];
+    totalDeposited: string;
+    totalBorrowed: string;
+}
+
+const LENDING_POOL_ADDRESS = "0xB702cE183b4E1Faa574834715E5D4a6378D0eEd3";
+const PROTOCOL_DATA_PROVIDER_ADDRESS =
+    "0x29563f73De731Ae555093deb795ba4D1E584e42E";
+const IUSD_ADDRESS = "0xA70266C8F8Cf33647dcFEE763961aFf418D9E1E4";
+
+export class IroncladService {
+    constructor() {}
+
+    @Tool({
+        name: "ironclad_loop_deposit",
+        description:
+            "Perform a looped deposit (recursive borrowing) on Ironclad. Send the amount of the asset (in base units) you want to deposit as the initial amount.",
+    })
+    async loopDeposit(
+        walletClient: EVMWalletClient,
+        parameters: LoopDepositParameters
+    ): Promise<LoopPosition> {
+        try {
+            console.log(
+                `[Ironclad] ====== Starting Loop Deposit Operation ======`
+            );
+            console.log(`[Ironclad] Asset Address: ${parameters.assetAddress}`);
+            console.log(
+                `[Ironclad] Initial Amount: ${parameters.initialAmount}`
+            );
+            console.log(`[Ironclad] Number of Loops: ${parameters.numLoops}`);
+            console.log(
+                `[Ironclad] User Address: ${walletClient.getAddress()}`
+            );
+
+            const position: LoopPosition = {
+                borrowedAmounts: [],
+                totalDeposited: "0",
+                totalBorrowed: "0",
+            };
+
+            const asset = parameters.assetAddress;
+            console.log(`[Ironclad] Checking allowance for initial deposit...`);
+            const allowanceResult = await walletClient.read({
```

Comment:
No need to have this, the LLM can figure it out or be prompted to do so. I would remove the whole ERC20 dependency here.